### PR TITLE
[CMake] Remove swift-syntax install component

### PR DESCRIPTION
**Pull request by harlanhaskins**
This is now unused.

### CODE REVIEW

**Comment by harlanhaskins** [comment]
@swift-ci please test

**Comment by shahmishal** [comment]
@swift-ci build toolchain

**Comment by harlanhaskins** [comment]
@swift-ci please build toolchain
@swift-ci please test

File: utils/build-presets.ini [hunk 1]
```diff
@@ -1070,7 +1070,7 @@ test-installable-package
 # If someone uses this for incremental builds, force reconfiguration.
 reconfigure
 
-swift-install-components=compiler;clang-builtin-headers;stdlib;swift-syntax;sdk-overlay;license;sourcekit-xpc-service;swift-remote-mirror;swift-remote-mirror-headers
+swift-install-components=compiler;clang-builtin-headers;stdlib;syntax;sdk-overlay;license;sourcekit-xpc-service;swift-remote-mirror;swift-remote-mirror-headers
```
[the remark this answers is not in the document]
Comment (harlanhaskins) [reply utils/build-presets.ini]: 🤦‍♂️

**Comment by harlanhaskins** [comment]
@swift-ci please test
@swift-ci please build toolchain

**Comment by Rostepher** [comment]
@swift-ci please test

**Comment by kastiglione** [comment]
single quotes in branch names? madness

**Comment by Rostepher** [comment]
Toolchain builds finished, not sure why Github hasn't realized it yet.

**Comment by kastiglione** [comment]
@harlanhaskins I'm not surprised git allowed it, surprised anyone would do that :trollface:

**Pull request merged**